webapp/views.py

```diff
@@ -393,6 +394,44 @@ def get_notice(notice_id, **kwargs):
             404,
         )
 
+    releases = {
+        release.codename: release.version
+        for release in db_session.query(Release).all()
+    }
+
+    package_descriptions = {}
+    package_versions = {}
+    release_packages = SortedDict()
+
+    if notice.release_packages:
```

Comment:
This block is certainly opaque to understand, takes quite a lot of thought. You could either add a largeish comment section explaining what's going on in plan language, or probably better would be to write an `add_related_notices` function, then you can give it a proper docblock:

``` python3
def add_related_notices(notice):
    """
    This does some magic
    """

    ... some magic

    return notice


def get_notice():
    ...

    return get_related_notices(notice)
```

Reply:
Then there's a further question of whether you leave this `_add_related_notices` function inside this file, or extract it into a separate file of `helpers` or `transformers` or something, for neatness (`from webapp.transformers import add_related_notices`). Maybe extracting it is overengineering? I'm not sure. Up to you.